packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -197,11 +370,98 @@ export class Table extends Resource {
       actions: ['dynamodb:ListStreams'],
       resourceArns: ['*'],
     });
- }
+  }
+
+  /**
+   * Creates a Table construct that represents an external table via table name.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param tableName The table's name.
+   */
+  public static fromTableName(scope: Construct, id: string, tableName: string): ITable {
+    return Table.fromTableAttributes(scope, id, { tableName });
+  }
+
+  /**
+   * Creates a Table construct that represents an external table via table arn.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param tableArn The table's ARN.
+   */
+  public static fromTableArn(scope: Construct, id: string, tableArn: string): ITable {
+    return Table.fromTableAttributes(scope, id, { tableArn });
+  }
+
+  /**
+   * Creates a Table construct that represents an external table.
+   *
+   * @param scope The parent creating construct (usually `this`).
+   * @param id The construct's name.
+   * @param attrs A `TableAttributes` object.
+   */
+  public static fromTableAttributes(scope: Construct, id: string, attrs: TableAttributes): ITable {
+
+    class Import extends TableBase {
+      public readonly tableName: string;
+      public readonly tableArn: string;
+
+      constructor(_scope: Construct, _id: string, props: TableAttributes) {
+        super(_scope, _id);
+        this.tableArn = props.tableArn!;
+        this.tableName = props.tableName!;
+      }
+
+      protected get hasIndex(): boolean {
+        return false;
+      }
+    }
+
+    const stack = Stack.of(scope);
+    let tableName = '';
+    let tableArn = '';
```

Comment:
Change this to `let tableArn: string;` (the compiler will help you catch all branches where this will not be initialized).